File: trakt-api/src/main/java/net/simonvt/cathode/api/ApiInterceptor.java
package net.simonvt.cathode.api;

import java.io.IOException;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class ApiInterceptor implements Interceptor {

  public static final String HEADER_API_KEY = "trakt-api-key";

  public static final String HEADER_API_VERSION = "trakt-api-version";

  private TraktSettings settings;

  public ApiInterceptor(TraktSettings settings) {
    this.settings = settings;
  }

  @Override public Response intercept(Chain chain) throws IOException {
    Request request = chain.request()
        .newBuilder()
        .addHeader(HEADER_API_KEY, settings.getClientId())
        .addHeader(HEADER_API_VERSION, "2")
        .build();
    return chain.proceed(request);
  }
}

File: trakt-api/src/main/java/net/simonvt/cathode/api/ApiInterceptor.kt
package net.simonvt.cathode.api

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ApiInterceptor(val settings: TraktSettings) : Interceptor {

  @Throws(IOException::class)
  override fun intercept(chain: Interceptor.Chain): Response {
    val request = chain.request()
      .newBuilder()
      .addHeader(HEADER_API_KEY, settings.clientId)
      .addHeader(HEADER_API_VERSION, "2")
      .build()
    return chain.proceed(request)
  }

  companion object {
    const val HEADER_API_KEY = "trakt-api-key"
    const val HEADER_API_VERSION = "trakt-api-version"
  }
}
